palette/moreland/luminance.go

```diff
@@ -0,0 +1,255 @@
+	}
+	c1 := l.colors[i-1]
+	c2 := l.colors[i]
+	frac := (scalar - l.scalars[i-1]) / (l.scalars[i] - l.scalars[i-1])
+	o := cieLAB{
+		L: frac*(c2.L-c1.L) + c1.L,
+		A: frac*(c2.A-c1.A) + c1.A,
+		B: frac*(c2.B-c1.B) + c1.B,
+	}.cieXYZ().rgb().sRGBA(l.alpha)
+	o.clamp()
+	return o, nil
+}
+
+// SetMax implements the palette.ColorMap interface for a luminance object.
+func (l *luminance) SetMax(v float64) {
+	l.max = v
+}
+
+// SetMin implements the palette.ColorMap interface for a luminance object.
+// However, it will panic whenever it is called because the
+// minimum value must always be zero.
+func (l *luminance) SetMin(v float64) {
+	panic("moreland: Luminance minimum value cannot be changed from zero")
+}
+
+// Max implements the palette.ColorMap interface for a luminance object.
+func (l *luminance) Max() float64 {
+	return l.max
+}
+
+// Min implements the palette.ColorMap interface for a luminance object.
+func (l *luminance) Min() float64 {
+	return 0
+}
+
+// SetAlpha sets the opacity value of this color map. Zero is transparent
+// and one is completely opaque.
+// The function will panic is alpha is not between zero and one.
+func (l *luminance) SetAlpha(alpha float64) {
+	if alpha > 1 || alpha < 0 {
+		panic(fmt.Errorf("invalid alpha: %g", alpha))
+	}
+	l.alpha = alpha
+}
+
+// Alpha returns the opacity value of this color map.
+func (l *luminance) Alpha() float64 {
+	return l.alpha
+}
+
+// Palette returns an object that fulfills the palette.Palette interface,
+// where nColors is the number of desired colors.
+func (l luminance) Palette(n int) palette.Palette {
+	if l.max == 0 {
+		l.max = 1
+	}
+	delta := l.max / float64(n-1)
+	v := 0.0
+	c := make([]color.Color, n)
+	for i := 0; i < n; i++ {
+		var err error
+		c[i], err = l.At(v)
+		if err != nil {
+			panic(err)
+		}
+		v += delta
+	}
+	return plte(c)
+}
+
+// plte fulfils the palette.Palette interface.
+type plte []color.Color
+
+// Colors fulfils the palette.Palette interface.
+func (p plte) Colors() []color.Color {
+	return p
+}
+
+// BlackBody is is Luminance-class ColorMap based on the colors of black body radiation.
+// Although the colors are inspired by the wavelengths of light from
+// black body radiation, the actual colors used are designed to be
+// perceptually uniform. Colors of the desired brightness and hue are chosen,
+// and then the colors are adjusted such that the luminance is perceptually
+// linear (according to the CIE LAB color space).
+func BlackBody() palette.ColorMap {
+	return &luminance{
+		colors: []cieLAB{
+			cieLAB{L: 0, A: 0, B: 0},
+			cieLAB{L: 39.112572747719774, A: 55.92470934659227, B: 37.65159714510402},
+			cieLAB{L: 58.45705480680232, A: 43.34389690857626, B: 65.95409116544081},
+			cieLAB{L: 84.13253643355525, A: -6.459770854468639, B: 82.41994470228775},
+			cieLAB{L: 100, A: 0, B: 0}},
+		scalars: []float64{0, 0.39112572747719776, 0.5845705480680232, 0.8413253643355525, 1},
+		alpha:   1}
```

Comment:
please add a comma after the `1` and put the closing brace on a new line (like you did in `ExtendedBlackBody`).
this reduces somewhat git-diff-linenoise.